cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -234,6 +235,11 @@ class LeadImpl extends ServerImpl with Lead
         markLauncherRunning(if (startupString ne null) s"Starting $startupString" else null)
       }
 
+      // Add a URL classloader to the main thread so that new URIs can be added
+      val parent = Thread.currentThread().getContextClassLoader
```

Comment:
Its a common practice to use Utils.getContextOrSparkClassLoader. You can use it for parent classloader.